src/remediations/remediations.format_2.js

```diff
@@ -52,52 +37,79 @@ exports.connectionStatus = function (recipients) {
     //   - all connected direct
     //   - all disconnected direct
     //   - all no-rhc direct
-    const data = _(recipients)
-        .groupBy(recipient => {
-            switch (recipient.recipient_type) {
-                case 'satellite':
-                    return `${recipient.sat_id} ${recipient.sat_org_id}`;
 
-                default:
-                    return recipient.recipient_type;
-            }
-        })
-        .map((targets) => {
-            const recipient = targets[0];
-            switch (recipient.recipient_type) {
-                case 'satellite':
-                    return {
-                        endpoint_id: null,
-                        executor_id: recipient.sat_id,
-                        executor_type: 'RHC-satellite',
-                        executor_name: `Satellite ${recipient.sat_id} Org ${recipient.sat_org_id}`,
-                        system_count: recipient.systems.length,
-                        connection_status: recipientStatusMap[recipient.status]
-                    };
-
-                case 'directConnect':
-                    return {
-                        endpoint_id: null,
-                        executor_id: null,
-                        executor_type: 'RHC',
-                        executor_name: null,
-                        system_count: targets.length,
-                        connection_status: computeStatus(targets)
-                    };
-
-                default:
-                    return {
-                        endpoint_id: null,
-                        executor_id: null,
-                        executor_type: 'None',
-                        executor_name: null,
-                        system_count: recipient.systems.length,
-                        connection_status: recipientStatusMap[recipient.status]
-                    };
-            }
-        })
-        .sortBy('executor_name')
-        .value();
+    const data = _(recipients)
+    .groupBy(recipient => {
+        // key will either be one of recipientStatusMap or a recipient_id UUID
+        switch (recipient.recipient_type) {
+            case PD_TYPE_DIRECT:
+                return recipient.status;
+
+            case PD_TYPE_SATELLITE:
+                return `${recipient.sat_id} ${recipient.sat_org_id}`;
+
+            default:
+                return PD_STATUS_NOT_CONFIGURED;
+        }
+    })
+    .map((items, group) => {
+        const system_ids = _(items).map('systems').flatten().value();
```

Comment:
Is this supposed to aggregate recipients of type direct by status?

Reply:
It's a slight semantic change, but not much of a departure from how it has functioned in the past.  This endpoint has evolved somewhat to accommodate first RHC-direct systems and then RHC-satellite systems.  It's undergone greater _syntactic_ changes in the past without creating a new endpoint version so we're pretty comfortable with this.  Also, it now actually provides useful information for direct systems.